### DISCUSSION TITLE
Can I use moddable SDK with a Lilygo v3 T-watch?

### DISCUSSION THREAD

**Discussion by haikyuu** [Moddable-OpenSource/moddable#1350] [Porting]
Here is a [link to the product](https://www.aliexpress.com/item/1005002053650442.html?gatewayAdapt=4itemAdapt) and to the [pinout table in github](https://github.com/Xinyuan-LilyGO/TTGO_TWatch_Library/blob/master/docs/watch_2020_v3.md)

Is it already supported? 

P.S: I'm a web developer and the Moddable book is still on the way 😅

**Comment by phoddie** [comment]
Welcome!

There's no doubt that the Moddable SDK can work with the Lily v3 T-watch. But, there's no port for it yet. So, that's some work.

Probably the best way to begin a port would be from an existing port that is similar. The existing ports are listed in our [ESP32 docs](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/devices/esp32.md#platforms) and the code and configurations for those are in our [GitHub repository](https://github.com/Moddable-OpenSource/moddable/tree/public/build/devices/esp32/targets). 

Looking at the pinout table you linked to, the display controller and touch controller are supported in the Moddable SDK, so that should just be configuration. But the backlight seems to be connected to a power management chip, which could be a bit of work to bring-up.

For an experienced embedded developer, that should be all be pretty straightforward. Coming from the web universe, it will likely be more challenging. We can use this thread to help with the details, if you do attempt that. But, for your initial explorations with the Moddable SDK (and our book, thank you!) I'd strongly recommend starting with a hardware device that is already supported.

**Reply by haikyuu** [reply]
Thanks for the detailed response.

I will start my project using the simulator then :) as I only have a [LilyGo eink display](https://usa.banggood.com/LILYGO-7_5-inch-E-ink-Display-Screen-Bare-Board-800x480-Compatible-With-T5-Motherboard-p-1871832.html?cur_warehouse=CN) with its [T5 motherboard](https://github.com/Xinyuan-LilyGO/LilyGo-T5-Epaper-Series) and It's probably a more complicated project for me right now 😅

When I discovered Moddable [here in this W3C spec](https://runtime-keys.proposal.wintercg.org/), I immediately knew I would use it to make my home made watch OS 🤩 

>  We can use this thread to help with the details, if you do attempt that

I would love to

**Reply by phoddie** [reply]
Sounds good. (Funny that you found us through the W3C runtime keys registry!)  Best of luck with your initial explorations.
